src/crypto/shelley/ShelleyWallet.js

```diff
@@ -267,6 +245,7 @@ export default class ShelleyWallet extends Wallet implements WalletInterface {
     this.version = data.version
     this.internalChain = AddressChain.fromJSON(data.internalChain)
     this.externalChain = AddressChain.fromJSON(data.externalChain)
+    this.rewardAddressHex = await this.internalChain.getRewardAddressHex()
```

Comment:
I'm confused. How is `getRewardAddressHex` is related to `internalChain`

Reply:
both the internal and external chains cache the public key and can derive the reward address. But probably it'll be less confusing to derive it right there explicitly.